server/src/main/java/org/opensearch/common/cache/service/NodeCacheStats.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.service;
+
+import org.opensearch.action.admin.indices.stats.CommonStatsFlags;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.cache.CacheType;
+import org.opensearch.common.cache.stats.ImmutableCacheStatsHolder;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+/**
+ * A class creating XContent responses to cache stats API requests.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class NodeCacheStats implements ToXContentFragment, Writeable {
+    // Use TreeMap to force consistent ordering of caches in API responses
+    private final TreeMap<CacheType, ImmutableCacheStatsHolder> statsByCache;
+    private final CommonStatsFlags flags;
+
+    public NodeCacheStats(TreeMap<CacheType, ImmutableCacheStatsHolder> statsByCache, CommonStatsFlags flags) {
+        this.statsByCache = statsByCache;
+        this.flags = flags;
+    }
```

Comment:
Use the `SortedMap` interface wherever possible? My bad, missed this in the first iteration! :(
In general, code to interfaces instead of concrete implementations wherever possible

Reply:
Ah, didn't know about SortedMap interface, had done Map originally to avoid having it be the concrete TreeMap. Will change.